cluster/src/main/java/org/apache/iotdb/cluster/coordinator/Coordinator.java

```diff
@@ -250,6 +258,20 @@ public TSStatus processPartitionedPlan(PhysicalPlan plan) throws UnsupportedPlan
     return forwardPlan(planGroupMap, plan);
   }
 
+  /**
+   * check if batch execution plan has privilege on any sg
+   *
+   * @param plan
+   * @return
+   */
+  private boolean checkPrivilegeForBatchExecution(PhysicalPlan plan) {
+    if (plan instanceof BatchPlan) {
+      return ((BatchPlan) plan).getResults().size() != plan.getPaths().size();
+    } else {
+      return true;
+    }
```

Comment:
IMO, If` ((BatchPlan) plan).getResults().size()>0`, which means some plan not have the privilege on some sg. 
I do not clearly know what's the above code mean.

Reply:
batch execution can operate on multiple storage groups, if auser has privilege on any of the sg, he can continue in processPartitionedPlan. on the other hand if he has no privilege on any sgs when getResults.size() == paths.size(). it will jump to concludeFinalStatus(...)

Reply:
ok